### ISSUE TITLE
Cmake test for Cocoa fails.

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL#1040]
# This bug report was migrated from our old Bugzilla tracker.

These attachments are available in the static archive:

* [small change to fix CHECK_OBJC_SOURCE_COMPILES (file_2056.txt, text/plain, 2013-08-27 00:59:38 +0000, 505 bytes)](https://bugzilla.libsdl.org/attachment.cgi?id=1307)
* [Patch for failing to detect Cocoa (macros.cmake.macosx.patch, text/plain, 2013-10-22 07:32:58 +0000, 532 bytes)](https://bugzilla.libsdl.org/attachment.cgi?id=1391)

**Reported in version:** 2.0.0
**Reported for operating system, platform:** Mac OS X 10.6, x86_64

# Comments on the original bug report:

On 2013-08-22 23:47:18 +0000,  wrote:

> This is related to a post I made on the forums a while back:
> 
> http://forums.libsdl.org/viewtopic.php?t=9100&highlight=
> 
> I finally got around to trying to troubleshoot this some more and managed to track down the cause of the linker issue being that simply all the files for Cocoa in the video subsystem weren't being added to the project.  Further digging showed it was failing the test that includes "Cocoa/Cocoa.h" to check if it is available.
> 
> On a random hunch I tried adding the compile flags to make it play nice with objective-c manually and it worked.  I made the changes to a copy of SDL in my local project repo and they can be seen here:
> 
> https://github.com/BlackToppStudios/Mezzanine/commit/c68e480ab6584b8e8b85ba9136d2c169bf92e9d5
> 
> I was a little over the top cautious regarding what was in "CMAKE_REQUIRED_FLAGS" at the time and that may not be necessary.  Also I didn't notice until after I made these changes that the function being used to test is "check_objc_source_compiles" and not "check_c_source_compiles" which raises some questions about that cmake methods stability/utilty.
> 
> TL;DR: Add "set(CMAKE_REQUIRED_FLAGS "-x objective-c")" prior to the cocoa check in SDL/cmake/sdlchecks.cmake around line 452 and all is good.

On 2013-08-27 00:59:38 +0000, Nathan Clack wrote:

> Created attachment 1307
> small change to fix CHECK_OBJC_SOURCE_COMPILES
> 
> This fixes Cocoa detection on my system (osx 10.8).  The test was failing due to a linker error; the objc runtime library wasn't being linked in.

On 2013-09-10 15:52:11 +0000, Timothee Besset wrote:

> Running into the same problem, Cocoa not detected on OSX 10.8
> 
> I applied the fix to the CHECK_OBJC_SOURCE_COMPILES macro that you attached, but the problem persists.

On 2013-10-22 07:32:58 +0000,  wrote:

> Created attachment 1391
> Patch for failing to detect Cocoa
> 
> I'm attached a patch that fixed the problem for me.

On 2014-02-23 03:30:47 +0000, Ryan C. Gordon wrote:

> 
> Taking this bug.
> 
> --ryan.

On 2014-02-23 06:12:16 +0000, Ryan C. Gordon wrote:

> 
> Fixed this in...
> 
>     https://hg.libsdl.org/SDL/rev/bfea1a568a0a
> 
> ...by removing the Objective-C test entirely. It's a Mac, it has Cocoa, let's not get crazy here.  :)
> 
> --ryan.
